udata/kafka/producer.py

```diff
@@ -0,0 +1,45 @@
+import json
+from flask import current_app
+from kafka import KafkaProducer
+from udata.models import db, Dataset, Organization, Reuse
+
+
+class KafkaProducerSingleton:
+    __instance = None
+
+    @staticmethod
+    def get_instance() -> KafkaProducer:
+        if KafkaProducerSingleton.__instance is None:
+            KafkaProducerSingleton.__instance = KafkaProducer(
+                bootstrap_servers=current_app.config.get('KAFKA_URI'),
+                value_serializer=lambda v: json.dumps(v).encode('utf-8')
+            )
+        return KafkaProducerSingleton.__instance
+
+
+def produce(model, id, message_type, document=None, index=None):
+    '''Produce message with marshalled document'''
+    producer = KafkaProducerSingleton.get_instance()
+    key = id.encode("utf-8")
+    if model == Dataset:
+        topic = 'dataset'
+    if model == Organization:
+        topic = 'organization'
+    if model == Reuse:
+        topic = 'reuse'
+    if not topic:
+        return
+
+    if not index:
+        index = topic
+    value = {
+        'service': 'udata',
+        'data': document,
+        'meta': {
+            'message_type': message_type.value,
+            'index': index
```

Comment:
I wonder, do you think we should set this meta property in kwargs? Because `index` is not generic to all messages produced.

Reply:
True. `message_type` either as its for now:
```
    INDEX = 'index'
    REINDEX = 'reindex'
    UNINDEX = 'unindex'
```
But I guess more types will be added in the future?

Reply:
Hmm `message_type` is generic and can be left in `produce` function I guess.
I meant the `index` key, or additional arbitrary meta keys.